### DISCUSSION TITLE
How do you use the Commit Activity page?

### DISCUSSION THREAD

**Discussion by rmathew1992** [community/community#40466] [Repositories]
### Select Topic Area

Product Feedback

### Body

👋 We'd love to know how you use the Commit Activity graphs on the Repository Insights tab (navigate to https://github.com/[REPO_NAME]/graphs/commit-activity), or click on "Insights" < "Commits" from your repository.

<img width="1613" alt="Screen Shot 2022-11-30 at 10 28 43 AM" src="https://user-images.githubusercontent.com/2595311/204856435-f9b3b32f-cb1a-4a2c-a09e-e6603ce94998.png">
<img width="1264" alt="Screen Shot 2022-11-30 at 11 41 45 AM" src="https://user-images.githubusercontent.com/2595311/204874533-61beca11-98ec-4458-9ade-8d6e0a58d3cd.png">

**Feedback we'd love to hear from you**

* How often do you use these graphs?
* Do you explore these graphs in your own repositories, or for other repositories?
* What data are you looking to get from it?
* What actions do you expect to take using the data from these graphs?
* What could we do to make this data easier to use or better for you?
* Anything else you think we should know?

**Comment by A8880** [comment]
im not intersted to get agione becu i have feil all the would know is katsh me

**Reply by A8880** [reply]
q1 ( at ladt 6 month i tiket like 6 taime
q3 ( never i will never take agine) 
q4 ( u know when u bot me under depursh and test like this i dont know how the grahp he come to me i think the ghraph is come wtih test but this is not first time for the tested i thinke the tested start when im in sudan befour when i into to hosptal to eiehav

**Comment by ghost** [comment]
The commits insight are really good, it helps visualize active days and average commits per day. I look at it in my personal repository and organisations 

However it's not mobile friendly 
![Screenshot_20221206_120144_Chrome](https://user-images.githubusercontent.com/37674069/205907099-fc5c108e-bafd-4749-9c87-aca2f1942717.jpg)

**Reply by dapsongbite** [reply]
Please I need my membership of the Zero To Mastery Organisation restored, I was removed because I didn't have 2 factors authentication, I have done it now @rmathew1992 @martinwoodward

**Reply by GwynethLlewelyn** [reply]
Clearly off-topic.

**Comment by 965022-oss** [comment]
I want to try action commit obtaining the high sampling rate on my API application
It has 60HZ merge to 200 HZ if it does.

**Comment by 965022-oss** [comment]
accurate delivery on my action commit.thank you for your kindly consideration.

**Comment by 965022-oss** [comment]
Write repository in your browser then make pull request with your document
file label the required needed prior to submission added the sort in
existing application you pick the necessary document like
build.gradle,plugins,dependencies,variables and so on.Next you must applied
it on your rest API if completed and you must have your Fix Google Play
services that create advance update if you received notification prior with
your request that correct.thank you and more power keep on doing the right
thing..install Github reliable in committed activity.

**Comment by ghost** [comment]
Ban these bots.....

**Comment by aldoBrett** [comment]
I would like to see this graphs by branch, right now you can only see it when everything has been merged to the main branch.

**Reply by davesrocketshop** [reply]
This. For all the features. Right now my project looks like I've done nothing for months when I've been committing almost every day. People checking the stats will see it as inactive just because I haven't merged to main.

**Reply by ksclarke** [reply]
Yes, this. I have one project where different versions are in different branches and `main` is just used as the landing page / general documentation page. I'd like to link to a page of committers, but they're not listed since only I have committed to the `main` branch.

**Reply by Christiana0000** [reply]
Great idea

**Comment by stephenstrenn** [comment]
The network view is much appreciated!

**Comment by Alex-302** [comment]
Hi.

I use it to track team activity.

> How often do you use these graphs?

every day

> Do you explore these graphs in your own repositories, or for other repositories?

own

> What data are you looking to get from it?

- Closed issues (total and by each team member)
- Created  issues (total and by each team member)
- New pull requests (total and by each team member)
- Merged pull requests (total and by each team member)
- Commits by each team member

> What could we do to make this data easier to use or better for you?

Split the statistics into general and separately for team members (select one at a time)

> Anything else you think we should know?

Punchcard was useful) Please restore it)

**Reply by piquark6046** [reply]
In additional, including each collaborator and contributor is a good idea.

**Reply by GwynethLlewelyn** [reply]
+1 on the punchcard

**Comment by slonik-az** [comment]
I use commit page to gain a quick overview of a project's commit activity. Very useful tool, indeed.

**Comment by simov** [comment]
> How often do you use these graphs?

Unfortunately never. The top chart that contains the commit counts per week is somewhat similar to what you can get in the Contributors section, which on its own contains way more useful information. And then the bottom chart that shows the commits per day in a week - I don't really find that useful either.

**Comment by coderedart** [comment]
I mostly use it to get a rough overview of the project's maintenance status.

A lot of projects just casually die off into the background without archiving the repo or mentioning the maintenance status in the README. and using insights on github tells me 
1. if anyone is still working on the project across all the branches (not just master)
2. the trend of the activity. like are the average commits per week increasing or decreasing, why they are high during certain timelines etc..
3. the contributors page also tells me whether its just a solo dev or whether a lot of volunteers are working on a project. 

Overall, these insights help me make some decisions with confidence like whether to use the project as a dependency etc..

**Reply by DanielGedamo** [reply]
I want to try action commit obtaining the high sampling rate on my API application
It has 60HZ merge to 200 HZ if it does

**Reply by zopieux** [reply]
This right there. The vast majority of open-source projects published on GitHub are in a limbo state because the author(s) got burned off, became a parent, died, or decided to do something else. This fact has little to do with a project's popularity.

Understanding this reality, the Commit activity page gives a pulse indicator of a project at a glance, without having to check activity on all branches individually, etc.

**Comment by makinghappen** [comment]
The graph could also show key committer avatars and have an option to compare periods, showing multiple weekly graphs.

**Comment by guitarpicva** [comment]
It is interesting to view how many folks have viewed or cloned the repo, though I am not sure if it monitors the cold storage clones.  It probably should NOT do that because they would skew the results.

I'd like to see a user level activity monitor in addition to the repository level.

**Comment by manoelcampos** [comment]
Trying to access Insights >> Commits just returns the error `We tried our best, but the graph wouldn’t load. Try reloading the page.` This is happening in different repositories for months.

**Reply by DanielGedamo** [reply]
Thanks, I tried it too and it partially helped

**Comment by MarkWhybird** [comment]
As someone who usually writes and sees dates in DD/MM order, I found the labels at the bottom of the code frequency graph I just saw really confusing, especially as they have no year mentioned that I could see. Maybe the month could be the three letter code instead? (i.e. Dec19 instead of 12/19.)

12/19
01/09
01/30
02/20
03/13
04/03
04/24
05/15
06/05
06/26
07/17
08/07
08/28
09/18
10/09
10/30
11/20
12/11

**Reply by ssbarnea** [reply]
Please follow ISO-8601! That means that `MM-DD`, `YY-MM` should be used in crowded places. Never go for slashes as you open the pandora box regarding US date format versus almost everyone else.

**Reply by GwynethLlewelyn** [reply]
Aye to all of the above. It confused me as well. I'd be fine with a 3-letter code for the month (it's visually fine and at least unambiguous). Better still: change format according to local preferences or browser preferences (I'm sure that info is hidden somewhere). If all else fails: aye, ISO-8601!

It wasn't obvious to me that the actual statistics are _per week_ and that those dates mean 'the week starting on ...' — which, in turn, is confusing for some, since not everybody can agree on which day the week starts: depending on your culture/religion, it might start on a Sunday (classic Christian view), on a Monday (secular view), on a Friday or Saturday (according to some Religions of the Book), or not having special significance altogether (e. g. Buddhism).

**Comment by 965022-oss** [comment]
965022-oss: If the reviewed has been done please add my username and email on your list included in your version 2/up-for-grabs.net  global version thank you more power.. #37191

**Comment by nenaqui** [comment]
Quisiera poder ver la rama

**Comment by cwayne18** [comment]
I use these quite often, but I think a nice new feature could be the ability to generate a yearly report at the end of the year for the repo, giving some high level stats such as 
* Number of commits/committers
* Number of commits/PRs from outside contributors
* Number of releases
* Number of issues opened/closed

**Comment by 965022-oss** [comment]
You have to used this for if merges have been accumulate or sync on devices
observe behavior changes on your phone or computer discussion has to be
done on clarification...

**Comment by 965022-oss** [comment]
I had to complete my pull request activity as soon as possible while you
are in the activity choose the data that can be used available to run on
your rest API if it does'nt try other document you should apply the right
input on the data that you want to install on the device there are other
option fix Google Play services check the variable that you can used  with
build.gradle services..

**Comment by GwynethLlewelyn** [comment]
I'd love to have a view for more than a year of commits — that's important for longer-term projects which have few commits on a weekly or even monthly basis. The punchcard view allowed that!

**Comment by ianderrington** [comment]
Please link to the commits that were made so we can better evaluate them for volume/quality etc.

**Comment by prchaganti** [comment]
There needs to be a longer timeline for the views!

**Comment by erzz** [comment]
My honest suggestion is that all the current insights are so basic that they don't really provide a great deal of value.

If you are serious about the next step in letting the data shine - just acquire linearb.io so we can work with contributor data that adds value in terms of VSM, Dev Workloads, even some basic DORA stuff. 

We basically pay those guys just to cover all the stuff that we would expect Github to provide OOTB to compete with Gitlab in this area

**Comment by schlagerVIZIO** [comment]
Would be nice to see the git reflog from the server's perspective either here or somewhere else to be able to recognize activity like force pushes/history modifications etc.

**Comment by Ishan2608** [comment]
I have been adding commits to my code and syncing it with my GitHub account through my IDE Visual Studio code. I made 10 commits. When I visited the GitHub account and created a ReadMe file, it is showing that I have made only a single commit. The 10 commits are shown in the name of GitHub. Thus, in insights, it shows that 2 authors have made contributions to this project, whereas, I am the only one doing it. This needs to be fixed. This is why, even though the repo has a larger number of commits than other repos, it is not showing on top in overview. It should show only 1 author, which is myself, since I am the one doing it. This problem never occurred when I make commits from PyCharm. The commits are always in the name of my GitHub account, and thus the repos with larger number of commits are on top.

**Comment by krtschmr** [comment]
At least i can see somewhat who's productive and who's less productive. It sums up over the time. quite toxic tho :)

**Reply by appliedrd** [reply]
download commits to csv
I have to use my python script and your API to do that

**Comment by DanL12186** [comment]
It would be great to either have changes to things like lockfiles excluded by default, or part of a list of excludable file types. One 20KLOC yarn file change can pretty much render the scale of the graph unreadable (even enormous 2K code contributions look tiny!).

**Comment by DanielGedamo** [comment]
Would be nice to see the git reflog from the server's perspective either here or somewhere else to be able to recognize activity like force pushes/history modifications etc.

**Comment by d-rideout** [comment]
Can you give us more information about the repository's clone history?
* Can you show the history back to the beginning of the repository, rather than just displaying the previous two weeks? 
* Can you show something more about *who* is cloning the repository?
* (Bonus round:) Can you superimpose the clone history with some measure of the commit history?  Such as number of commits each day?

**Comment by nickman** [comment]
Would really like to see stats on release artifact downloads.

**Comment by BeZamorano** [comment]
**How often do you use these graphs?**
- Monthly, I need this data at least once per month (generally is more).
**Do you explore these graphs in your own repositories, or for other repositories?**
- Organisation all around.
**What data are you looking to get from it?**
- Understand how our teams are working, # commits per PR, status, etc.
**What actions do you expect to take using the data from these graphs?**
- Define best practices, improve on stuff we might be not working properly, etc.
**What could we do to make this data easier to use or better for you?**
- Having more filters, being able to select data ranges, being able to see it **per month** rather than per week
- Are the squared commits count as one? If so - I'd like to have the option to have a count of all the commits even the ones squashed.

**Comment by JeremyDTaylor** [comment]
I'd love quarterly (3 months) graphs and statistics

**Comment by richelbilderbeek** [comment]
Me and my team would love to see the total number of commits as well, as then we can celebrate our -for example- ten-thousanth commit :partying_face:

**Reply by vatrofi** [reply]
The network view is much appreciated!

**Comment by corwinn** [comment]
> How often do you use these graphs?

Rare.
> Do you explore these graphs in your own repositories, or for other repositories?

Both. To compare.
> What data are you looking to get from it?

Learn the difference between commit for the good looks of "a lot of activity", or because its actually required; some employers think "commit(s) each day" means skill. Code quality != Code quantity, in my humble POV.
> What actions do you expect to take using the data from these graphs?

Compare yours with my own code-metrics. Code-metrics tend to make me improve my code.
> What could we do to make this data easier to use or better for you?

Its fine. Thank you.
> Anything else you think we should know?

Thats not for me to answer.

**Comment by jchou1271** [comment]
Can you add a commit activity page filter so that you can see commit counts per branch?  I assume the weekly counts displayed currently is across all branches.

**Comment by d-rideout** [comment]
Is there an API interface to this information?  (In particular the clones and webpage visits.)  Can we just code our own Insights tab?   Or perhaps two: one public facing and one for our own information?  

This would be great, and not require agreement among many diverse opinions and goals.

**Comment by chiangbi** [comment]
with openwrt

**Comment by floede** [comment]
Pretty useless when I can't even adjust the time period.

**Comment by DanielGedamo** [comment]
Pretty useless when I can't even adjust the time period.

**Comment by cube-drone** [comment]
> How often do you use these graphs?
Rarely

> Do you explore these graphs in your own repositories, or for other repositories?
My own repositories, very rarely when I'm trying to get a feel for how active a project is (I usually check tickets/commits instead)

> What data are you looking to get from it?

* Vanity metrics ( "who's committing those big juicy refactorings? is it me?")
* Mean/median/stddev time from PR request to "Accepted" review 
* Mean/median/stddev time from "Accepted" to merge
* PR size / relationship between PR size and time-to-merge
* basically: a lot of the stuff provided by monocle ( https://demo.changemetrics.io/python ) 
* While I don't use them, I imagine people would want similar metrics to these for tasks & tickets

> What actions do you expect to take using the data from these graphs?

* I want to make the numbers go down, using magic or by cheating
* I want to encourage people to make more digestible PRs 

> What could we do to make this data easier to use or better for you?

* It'd be nice if it had some more of the data I'm looking for.

**Comment by Slackadays** [comment]
I love using the Traffic and Network sections to monitor how much popularity there actually is for my project. Unfortunately, I'd like if the Commits section didn't just cover one week in the day graph, but rather maybe an average of the last few weeks. I also noticed that even on my workstation with a Ryzen 5900x and 6800XT, the Forks tab freezes my browser if there's too much to display, which is the case for most popular projects.

**Comment by rohitnandi12** [comment]
What data are you looking to get from it?
I want to navigate through different commits.

What actions do you expect to take using the data from these graphs?
On clicking the dots in the weekly graph; that commit should open in a new tab.

**Comment by rdkls** [comment]
I almost always use these graphs for assessing freshness/activity of development and support, as a datapoint in selecting a tool/framework.

Major feature missing for me in this case, is ability to view older times (prior to one month)

Further for this use case, a summary of a repo's dependencies (and health thereof - supply chain and all that), accompanying security issues

**Comment by Cornelius-Figgle** [comment]
slow

**Comment by geira** [comment]
Seems like it's only possible to extract history for the last year. We want stats for commits at least 10 years ago.

**Comment by st3fan** [comment]
Maybe this is an edge case but I ran into it recently: I wanted to know if work was being done on a project on the weekend. When you select a week you can get to this data but I _think_ it is presented from my timezone perspective. Which means that if you have a distributed team across timezones with big time differences, I will see for example commits from Australia appear in the count for Sunday.

This is difficult in UX probably, but maybe a "From the committers perspective" kind of checkbox would be interesting.

**Comment by jenreiher-eb** [comment]
Context: I am the PM of the design ops team building a design system. I am tasked with collecting data to tell the story of how other teams adopt and use our system (aka "adoption"), and their git history is a great data point in that for us. 

How often do you use these graphs?
- This week was my first time

Do you explore these graphs in your own repositories, or for other repositories?
- Other teams repositories, but still within my organization

What data are you looking to get from it?
- What is the number of commits per month / week a team makes, and is that up or down this month over the average and/or a past moment in time

What actions do you expect to take using the data from these graphs?
- In this case, I was coming to understand how to look at the number of commits overall to normalize it against our adoption. IE the number of commits in a giving time period, because if teams are committing more, that means we have likely more adoption happening. If teams are committing less, we have less opportunity for adoption. This makes sure that when do have less adoption due to seasonal changes in our work as a company I don't have to defend why the team isn't growing adoption as fast. 

What could we do to make this data easier to use or better for you?
- would have loved to be able to show the data by month or a rolling 30/60/90 day cycle at least  (google analytics does this well) 
- would have loved to be able to compare a set of time to the previous 30/60/90 days to see if the commits are a % up or down

Anything else you think we should know?
Here is the manual work i did in Excel with this data to visualize/catalog the information I needed.
<img width="1106" alt="Screen Shot 2023-01-27 at 10 42 02 AM" src="https://user-images.githubusercontent.com/99852792/215168715-e514cc5c-0311-404f-92fb-be5a8701f46c.png">

**Comment by ptesarik** [comment]
- How often do you use these graphs?
  - When pondering contribution to a project that I haven't been involved in, I can get a quick estimate of how active it is, and whether it is more likely a professional project (most commits on weekdays), or a hobby project (most commits at weekends).
- Do you explore these graphs in your own repositories, or for other repositories?
  - I sometimes have a look at my own repositories out of curiosity, but these graphs are more useful for other repositories.
- What data are you looking to get from it?
  - Days of week (weekday/weekend), average and variance in the number of commits per week.
- What actions do you expect to take using the data from these graphs?
  - It gives me an idea of how long it might take to merge my pull requests. Especially, if some comments and additional requests from the maintainer are expected, I can plan my time either as a short-term intensive interaction, or a long-term low-bandwidth one.
- What could we do to make this data easier to use or better for you?
  - GitHub might assign each project to a category (or multiple categories) based on its development pace, so I could immediately see what to expect. But I guess such division is quite subjective, and you don't want to fight even more flamewars, because people don't like their project's categories, or their names, or… ;-)
- Anything else you think we should know?
  - Nope.

**Comment by manvillej** [comment]
I am mainly looking at how active a project is. If I am thinking of using tool X over tool Y, I'd rather choose the more active project.

It would be interesting to have the ability to compare projects across these metrics. If I could select Django and Flask, then see a comparison of bug reports, PRs, commits, projects with them as a dependency. maybe see similar projects or other common comparisons.

**Comment by upyr1** [comment]
I was wonderign if how to run this locally?

**Comment by dirkx** [comment]
Useful to get a longer term view (too) -- to gauge overall health of the community (and thus how depending you can be on this code).

**Comment by KazuDante89** [comment]
Its a good feature overall . One thing that would make it better on github is to display wich branch had changes made to it recently because those of us that have tons of branches sometime we forget wich branch is the most recent ( also would come in handy for archived repos with several branches that we have not touched in months/years in case of a refresher to know where we left-off )

**Comment by borilski** [comment]
It is good vizialization of activity. It would help me, if I could change the visualized timeline - day, week, month, year and if I could see also how many users did these commits, not only the number of the commits. It is helping to compare the activity with past period and to identify changes in the activity growth.

**Comment by dpolombodkt** [comment]
It would be lovely to see the year when hovering over the bars, and be able to go back farther than a year.

**Comment by charlieforward9** [comment]
Hello! I am a heavy user of the GitHub Contributions graph. I use it to "gamify" my projects, and set a goal to make "one contribution per day" to get my green box. It would be cool to have this on the project level, for each member of the team, etc. In short, I am looking for more ways that the GitHub data could be gamified. 

I could spend time thinking of ideas, but I think with a prompt from this perspective, the GitHub team could conduct some R&D on **gamification** and come up with some awesome new product lineups.  If you'd like more elaboration, reach out!

Thanks for asking this great question!

**Comment by jlord** [comment]
👋 Thanks so much for this feedback, everyone! It's very helpful for the planning we're kicking off now ✨
